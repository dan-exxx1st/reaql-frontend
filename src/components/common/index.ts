export { default as DialogHeader } from './DialogHeader';
export { default as ContactItem } from './ContactsItem';
export { default as UserHeader } from './UserHeader';
export { default as DialogTextField } from './DialogTextField';
export { default as MessageItem } from './MessageItem';
export { default as DialogHeaderBackdrop } from './DialogHeaderBackdrop';
export { default as MessageSearch } from './MessageSearch';
