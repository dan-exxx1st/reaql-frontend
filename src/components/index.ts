export { default as DialogListItem } from './common/DialogListItem';
export { default as EmojiPicker } from './common/EmojiPicker';
