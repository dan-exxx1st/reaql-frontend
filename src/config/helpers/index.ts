import { useState, useEffect } from 'react';

export function useLoaded({ src }: { src?: HTMLImageElement['src'] }) {
	const [loaded, setLoaded] = useState('false');

	useEffect(() => {
		if (!src) {
			return undefined;
		}

		setLoaded('loading');

		let active = true;
		const image = new Image();
		image.src = src;
		image.onload = () => {
			if (!active) {
				return;
			}
			setLoaded('loaded');
		};
		image.onerror = () => {
			if (!active) {
				return;
			}
			setLoaded('error');
		};

		return () => {
			active = false;
		};
	}, [src]);

	return loaded;
}
