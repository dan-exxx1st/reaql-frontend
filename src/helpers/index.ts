import { parse } from 'querystring';

export function getDialogIdFromSearch(search: string) {
    const params = parse(search);

    const dialogId = Object.entries(params)
        .map(([key, value]) => key.indexOf('dialog') !== -1 && value)
        .toString();

    return dialogId;
}
