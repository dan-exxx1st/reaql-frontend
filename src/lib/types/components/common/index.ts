import Theme from 'helpers/styled';
import { Dialog, Maybe, Message, Message_Statuses } from 'lib/graphql/types';
import { Dispatch, SetStateAction } from 'react';

export interface IUserHeaderProps {
    className?: string;
    setSearchUserOpened?: (val: boolean) => void;
}

export interface IMessageSearchProps {
    onClick?: () => void;
    value?: string;
    messageFilter?: (value: string) => void | Dispatch<SetStateAction<string>>;
}

export interface IMessageItemProps extends Message {
    className?: string;
}

export interface IDialogsItemProps {
    id: string;
    name: string;
    surname: string;
    avatar: string;
    unreadMessages: number | null | undefined;
    lastMessageStatus: Message_Statuses | null | undefined;
    lastMessage: string | null | undefined;
    lastMessageDate: string | null | undefined;
}

export interface IMessageListProps {
    data?: {
        messages: Message[];
    };
    loading?: boolean;
    subscribeToNewMessages?: () => void;
    className?: string;
    filterValue?: string;
}

export interface IDialogListProps {
    dialogFilter: string;
    dialogs?: Maybe<Dialog[]>;
    className?: string;
}

export interface IDialogTextFieldProps {
    className?: string;
}

export interface IDialogHeaderProps {
    className?: string;
    messageFilter?: Dispatch<SetStateAction<string>>;
    filterValue?: string;
    name: string;
    onlineStatus?: string | null;
    avatar?: string;
    loading?: boolean;
}

export interface ISignUpFormProps {
    email: string;
    name: string;
    surname: string;
    password: string;
    changeFormValues: (field: string, value: string) => void;
    handleSubmit: () => void;
    theme?: typeof Theme;
}

export interface ISignInFormProps {
    _handleSignIn: (payload: {
        email: string;
        password: string;
        rememberMe: boolean;
    }) => void;
}

export interface ISideBarProps {
    data?: {
        dialogs: Dialog[];
    };
    loading: boolean;
    subscribeToNewDialogs?: () => void;
    setSearchUserOpened: (val: boolean) => void;
}
