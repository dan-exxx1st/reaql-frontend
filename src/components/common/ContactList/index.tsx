import React, { FC } from 'react';

import { IContactListProps } from 'lib/types/components/common';

import { StyledContactListWrapper, StyledContactListItem } from './style';

const ContactList: FC<IContactListProps> = ({ contacts }) => {
    return (
        <StyledContactListWrapper>
            {contacts &&
                contacts.map((contact) => (
                    <StyledContactListItem key={contact.id} {...contact} />
                ))}
        </StyledContactListWrapper>
    );
};

export default ContactList;
