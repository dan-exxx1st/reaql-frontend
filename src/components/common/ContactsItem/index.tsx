import React, { FC, useEffect, useState } from 'react';
import StyledDialogListItem, {
    StyledContactsItemAvatarWrapper,
    StyledContactsItemCircle,
    StyledContactsItemBottomLine,
} from './style';
import { Wrapper, Avatar, Typography, CheckMark } from 'components/UI';
import { IContactsItemProps } from 'lib/types/components/common';

const ContactItem: FC<IContactsItemProps> = ({
    avatar_src,
    name,
    last_message,
    last_message_date,
    unread_message_count,
    sended_message_status,
}) => {
    const [isDoubleCheckMark, setIsDoubleCheckMark] = useState(false);
    const [isActiveCheckMark, setIsActiveCheckMark] = useState(false);
    const [undreadMessageCountText, setUnreadMessageCountText] = useState('');

    useEffect(() => {
        switch (sended_message_status) {
            case 'sended': {
                setIsDoubleCheckMark(false);
                setIsActiveCheckMark(false);
                break;
            }

            case 'received': {
                setIsDoubleCheckMark(true);
                break;
            }

            case 'readed': {
                setIsActiveCheckMark(true);
            }
        }
        if (unread_message_count) {
            if (unread_message_count > 0 && unread_message_count < 10) {
                setUnreadMessageCountText(unread_message_count.toString());
            } else {
                setUnreadMessageCountText('9+');
            }
        }
    }, []);

    return (
        <StyledDialogListItem flexWrap="wrap">
            <Wrapper height="40px" width="100%" justifyContent="space-between">
                <Wrapper height="40px">
                    <StyledContactsItemAvatarWrapper>
                        <Avatar src={avatar_src} alt="Avatar image" />
                    </StyledContactsItemAvatarWrapper>
                    <Wrapper
                        height="40px"
                        flexDirection="column"
                        justifyContent="space-between"
                    >
                        <Typography variant="body1" color="dc">
                            {name}
                        </Typography>
                        <Typography variant="caption1" color="dsc">
                            {last_message}
                        </Typography>
                    </Wrapper>
                </Wrapper>
                <Wrapper
                    flexDirection="column"
                    height="40px"
                    justifyContent="space-between"
                    alignItems="flex-end"
                >
                    <Typography variant="caption1" color="dgc">
                        {last_message_date?.toString()}
                    </Typography>

                    <Wrapper>
                        {undreadMessageCountText ? (
                            <Wrapper>
                                <StyledContactsItemCircle
                                    color="primary"
                                    text={undreadMessageCountText}
                                />
                            </Wrapper>
                        ) : (
                            <Wrapper>
                                <CheckMark
                                    isDouble={isDoubleCheckMark}
                                    isChecked={isActiveCheckMark}
                                />
                            </Wrapper>
                        )}
                    </Wrapper>
                </Wrapper>
            </Wrapper>

            <StyledContactsItemBottomLine />
        </StyledDialogListItem>
    );
};

export default ContactItem;
