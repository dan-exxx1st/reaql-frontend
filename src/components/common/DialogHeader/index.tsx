import React, { FC, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useQuery } from '@apollo/client';
import { decode } from 'querystring';

import { Typography, Wrapper } from 'components/UI';
import { MessageSearch } from '..';

import { StyledAvatar, StyledCircle, StyledDialogHeader } from './style';

import { IDialogHeaderProps } from 'lib/types/components/common';
import { Query } from 'lib/graphql/types';
import { GET_DIALOG } from 'lib/graphql/queries/dialog';

const DialogHeader: FC<IDialogHeaderProps> = ({ className }) => {
    const [isShowMessageSearch, setIsShowMessageSearch] = useState(false);

    const history = useHistory();
    const dialogId = decode(history.location.search)['?dialog'];

    const { data } = useQuery<Query>(GET_DIALOG, {
        variables: { dialogId },
    });

    const _handleToggleMessageSearch = () => {
        setIsShowMessageSearch(!isShowMessageSearch);
    };

    if (data && data.dialog) {
        const {
            dialog: { users, group },
        } = data;
        const name =
            !group && users[0]
                ? `${users[0].name} ${users[0].surname}`
                : 'group!!!'; //!SERVER
        const avatar = !group && users[0] ? users[0].avatar : ''; //!SERVER

        return (
            <StyledDialogHeader
                className={className}
                justifyContent="space-between"
            >
                {isShowMessageSearch ? (
                    <MessageSearch onClick={_handleToggleMessageSearch} />
                ) : (
                    <>
                        <Wrapper>
                            <StyledAvatar src={avatar} />
                            <Wrapper
                                flexDirection="column"
                                justifyContent="space-between"
                                height="40px"
                            >
                                <Typography variant="body1" color="dc">
                                    {name}
                                </Typography>
                                <Typography variant="caption1" color="dgc">
                                    Today at 3:30 pm
                                </Typography>
                            </Wrapper>
                        </Wrapper>
                        <Wrapper>
                            <StyledCircle
                                color="secondary"
                                iconName="search"
                                onClick={_handleToggleMessageSearch}
                            />
                            <StyledCircle
                                color="secondary"
                                iconName="threeDotsGrey"
                            />
                        </Wrapper>
                    </>
                )}
            </StyledDialogHeader>
        );
    }

    return <div>Loading...</div>;
};

export default DialogHeader;
