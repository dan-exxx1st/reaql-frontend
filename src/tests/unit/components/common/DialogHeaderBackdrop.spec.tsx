import React from 'react';
import { DialogHeaderBackdrop } from 'components/common';
import { mountWithTheme } from 'tests/helpers/withTheme';

describe('<DialogHeaderBackdrop />', () => {
    describe('Snapshots', () => {
        it('Should render correctly', () => {
            const wrapper = mountWithTheme(<DialogHeaderBackdrop />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
