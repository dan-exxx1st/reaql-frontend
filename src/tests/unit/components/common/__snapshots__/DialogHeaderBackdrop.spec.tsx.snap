// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogHeaderBackdrop /> Snapshots Should render correctly 1`] = `
.c5 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  color: #000000;
  margin: 0;
}

.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: auto;
  height: auto;
}

.c1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #E5E5E5;
  max-width: 200px;
  border-radius: 5px;
}

.c4 {
  margin-right: 10px;
}

.c6 {
  color: #00ADFF;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  width: calc(100% - 15px);
  padding: 5px 5px 5px 10px;
}

.c3:hover {
  background: #f5f5f5;
}

.c2 {
  position: absolute;
  top: -12px;
  right: 12px;
  width: 0;
  height: 0;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-bottom: 12px solid #E5E5E5;
}

<div
  className="c0 c1"
>
  <img
    className="c2"
  />
  <div
    className="c0 c3"
  >
    <img
      className="c4"
      src="BluePlus.svg"
    />
    <p
      className="c5 c6"
    >
      Add members
    </p>
  </div>
  <div
    className="c0 c3"
  >
    <img
      className="c4"
      src="BlueSearch.svg"
    />
    <p
      className="c5 c6"
    >
      Search message history
    </p>
  </div>
  <div
    className="c0 c3"
  >
    <img
      className="c4"
      src="BlueTrash.svg"
    />
    <p
      className="c5 c6"
    >
      Clear message history
    </p>
  </div>
</div>
`;
